src/components/calendar/MonthCalendar/MonthCalendarEvent.tsx

```diff
@@ -0,0 +1,79 @@
+import { FormattedTime } from 'react-intl';
+import { getContrastColor } from '../../../utils/colorUtils';
+import { grey } from '@material-ui/core/colors';
+import NextLink from 'next/link';
+import { ArrowBackIos, ArrowForwardIos } from '@material-ui/icons';
+import { Box,  Link, Typography } from '@material-ui/core';
+import { useEffect, useRef, useState } from 'react';
+
+import { getNaiveDate } from '../../../utils/getNaiveDate';
+import { ZetkinCampaign, ZetkinEvent } from '../../../types/zetkin';
+
+const DEFAULT_COLOR = grey[900];
+
+interface MonthCalendarEventProps {
+    baseHref: string;
+    isVisible: boolean;
+    campaign?: ZetkinCampaign;
+    event: ZetkinEvent;
+    startOfDay: Date;
+    onLoad?: (listItemHeight: number) => void;
+}
+
+const MonthCalendarEvent = ({ baseHref, startOfDay, campaign, event, onLoad, isVisible }: MonthCalendarEventProps): JSX.Element => {
+    const eventDiv = useRef<HTMLDivElement>(null);
+    const [focused, setFocused] = useState(false);
+
+    useEffect(() => {
+        if (eventDiv.current && onLoad) {
+            const listItemHeight = eventDiv.current.offsetHeight || 0 * 1.5;
+            onLoad(listItemHeight);
+        }
+    }, [onLoad]);
+
+    const startTime = getNaiveDate(event.start_time);
+    const endTime = getNaiveDate(event.end_time);
+
+    const endOfDay = new Date(new Date(startOfDay)
+        .setDate(startOfDay.getDate() + 1));
+
+    const startsBeforeToday = startTime <= startOfDay;
+    const endsAfterToday = endTime >= endOfDay;
+
+    return (
+        <li key={ event.id }>
+            <NextLink href={  baseHref + `/calendar/events/${event.id}` } passHref>
+                <Link underline="none">
+                    <Box
+                        onMouseEnter={ () => setFocused(true) }
+                        onMouseLeave={ () => setFocused(false) }
+                        { ... { ref: eventDiv }  }
```

Comment:
Now that there is no conditional logic here, this construct is unnecessarily complex. Could just replaced with `ref={ eventDiv }`, couldn't it?

Reply:
it's a bug in MUI -- hopefully fixed in the new version. Have used this workaround in several places in the codebase because Box has no 'ref' explicitly declared in its interface -- 
https://github.com/mui-org/material-ui/issues/17010